design/specs/visual-query-builder.md

```diff
@@ -0,0 +1,236 @@
+
+## 3. Applying Transformations to the Result Table
+
+The result table refers to the resulting table from all input columns selected and added, including filters and aggregations. Transforming this output can be done by applying output filters, sorts, and summarizations to the result table.
+
+### 3.1. Filtering the Output Table
+
+Users can filter the output table by adding a filter step to any column from the result table. The column selector, in this scenario, will only allow users to select input columns rather than the whole column list from the column selection stage.
+
+---
+Wireframes
+
+[Output Filter](https://share.balsamiq.com/c/dAVYnp8VnG2r2HzkSZ3rgp.png)
+
+### 3.2. Summarizing the Result Table
+
+The `Summarization` step enables users to generate a summarized version of the result table. When added, the system will group the table values based on a summary column of their choice. The summary can be done using the summary column's exact values or by using the summary column's data type's available grouping options.
+
+To summarize the table, the user selects the `Summarize` option from the result transformations menu and adds the step. Once added, the remaining columns are automatically assigned an aggregation function, which users can change at any time. The data type of the output column is used to infer aggregates.
+
+---
+Wireframes
+
+[Output Summarization](https://share.balsamiq.com/c/rPMwwETnuQ8a4ut8NfmcDB.png)
+
+[Summarization Options](https://share.balsamiq.com/c/x2iwVBT3NwzmKyhnxSiJku.png)
+
+### 3.3 Sorting the Output Table
+
+Users can sort the output table by applying a sort to any of the result table columns.
+
+To sort the table, the user selects the `Sort` option from the result transformations menu and adds it to the list. Once added, the user can set a column and a direction for the sort.
+
+---
+Wireframes
+
+[Output Sorting](https://share.balsamiq.com/c/8TYP1XNz49tS7hHqujMmqS.png)
+
+### 3.4 Adding a New Column to a Summarized Table
+
+If the result table has transformations applied, new columns will be added automatically to the summarization steps and an aggregation set by default.
+
+---
+Wireframes
+
+[Adding New Column](https://share.balsamiq.com/c/kWyvRL822BdBubhT4ghzgA.png)
+
+## 4. Deleting a Result Transformation Step
+
+To delete a result transformation step, the user can click on the `Delete` button present in each step item. Deleting a step means that the system will undo all transformations made through that step, and the result table will reflect the updated output.
+
+In circumstances where subsequent steps rely on columns resulting from a deleted step, the system will display error warnings for every failed step. The table output will only reflect the output generated by those stages without errors.
+
+---
+Wireframes
+
+[Deleting a Transformation Step](https://share.balsamiq.com/c/XeNAKGz1UnDMfscSTLDVp.png)
+
+## 5. Previewing the Query Results
```

Comment:
I think the preview needs more screen space. It would be better if we can explore options to merge both panes into one.

Reply:
I don't think merging both panes is a good idea. The left pane represents the structure of the query, the right pane is an "inspector" focusing on context-specific details for the current task the user is doing. 

I do agree that the user might want more space for the preview. Making both panes collapsible seems like a better option for this.